satpy/writers/geotiff.py

```diff
@@ -76,7 +76,27 @@ class GeoTIFFWriter(ImageWriter):
                     "profile",
                     "bigtiff",
                     "pixeltype",
-                    "copy_src_overviews",)
+                    "copy_src_overviews",
+                    # Not a GDAL option, but allows driver='COG'
+                    "driver",
```

Comment:
I would make this an explicit option on the save method and remove it from this list.